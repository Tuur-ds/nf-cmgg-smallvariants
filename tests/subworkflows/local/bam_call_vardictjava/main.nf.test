nextflow_workflow {

    name "Test Workflow BAM_CALL_VARDICTJAVA"
    script "subworkflows/local/bam_call_vardictjava/main.nf"
    workflow "BAM_CALL_VARDICTJAVA"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "bam_call_vardictjava"
    tag "vcf_concat_bcftools"
    tag "vcf_filter_bcftools"

    test("bam_call_vardictjava - default") {


        when {
            params {
                callers = "vardict"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143.00001", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split1, checkIfExists:true)
                ],[
                    [id:"NA24143.00002", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split2, checkIfExists:true)
                ],[
                    [id:"NA24143.00003", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split3, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists:true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists:true)
                ])
                input[3] = [[],[]]
                input[4] = [[],[]]
                input[5] = false
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs.collect { [it[0], "${file(it[1]).name},variantsMD5:${path(it[1]).vcf.variantsMD5}", file(it[2]).name] }
                ).match("default") }
            )
        }

    }

    test("bam_call_vardictjava - filter") {


        when {
            params {
                filter = true
                callers = "vardict"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143.00001", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split1, checkIfExists:true)
                ],[
                    [id:"NA24143.00002", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split2, checkIfExists:true)
                ],[
                    [id:"NA24143.00003", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split3, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists:true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists:true)
                ])
                input[3] = [[],[]]
                input[4] = [[],[]]
                input[5] = true
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs.collect { [it[0], "${file(it[1]).name},variantsMD5:${path(it[1]).vcf.variantsMD5}", file(it[2]).name] }
                ).match("filter") }
            )
        }

    }

    test("bam_call_vardictjava - family") {
        // The family should not be merged here

        when {
            params {
                callers = "vardict"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143.00001", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143,NA24149", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split1, checkIfExists:true)
                ],[
                    [id:"NA24143.00002", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143,NA24149", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split2, checkIfExists:true)
                ],[
                    [id:"NA24143.00003", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143,NA24149", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split3, checkIfExists:true)
                ],[
                    [id:"NA24149.00001", sample:"NA24149", family:"Ashkenazim", family_samples:"NA24143,NA24149", split_count:3],
                    file(params.bam2, checkIfExists:true),
                    file(params.bai2, checkIfExists:true),
                    file(params.split1, checkIfExists:true)
                ],[
                    [id:"NA24149.00002", sample:"NA24149", family:"Ashkenazim", family_samples:"NA24143,NA24149", split_count:3],
                    file(params.bam2, checkIfExists:true),
                    file(params.bai2, checkIfExists:true),
                    file(params.split2, checkIfExists:true)
                ],[
                    [id:"NA24149.00003", sample:"NA24149", family:"Ashkenazim", family_samples:"NA24143,NA24149", split_count:3],
                    file(params.bam2, checkIfExists:true),
                    file(params.bai2, checkIfExists:true),
                    file(params.split3, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists:true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists:true)
                ])
                input[3] = [[],[]]
                input[4] = [[],[]]
                input[5] = false
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs.collect { [it[0], "${file(it[1]).name},variantsMD5:${path(it[1]).vcf.variantsMD5}", file(it[2]).name] }
                ).match("family") }
            )
        }

    }

}
