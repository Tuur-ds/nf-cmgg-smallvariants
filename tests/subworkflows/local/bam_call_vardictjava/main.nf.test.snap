{
    "filter": {
        "content": [
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "caller": "vardict"
                    },
                    "NA24143.vardict.vcf.gz,variantsMD5:98497d2c15c6e3781f5ddeb81bf6288f",
                    "NA24143.vardict.vcf.gz.tbi"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-23T11:17:08.827619348"
    },
    "default": {
        "content": [
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "caller": "vardict"
                    },
                    "NA24143.vardict.vcf.gz,variantsMD5:98497d2c15c6e3781f5ddeb81bf6288f",
                    "NA24143.vardict.vcf.gz.tbi"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-23T11:16:29.818590197"
    },
    "family": {
        "content": [
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "caller": "vardict"
                    },
                    "NA24143.vardict.vcf.gz,variantsMD5:98497d2c15c6e3781f5ddeb81bf6288f",
                    "NA24143.vardict.vcf.gz.tbi"
                ],
                [
                    {
                        "id": "NA24149",
                        "sample": "NA24149",
                        "family": "Ashkenazim",
                        "family_samples": "NA24149",
                        "caller": "vardict"
                    },
                    "NA24149.vardict.vcf.gz,variantsMD5:37c24a3165a79df0b9744e80a9255f83",
                    "NA24149.vardict.vcf.gz.tbi"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-23T11:17:59.686967502"
    }
}